kafka/signed_token_redeem_handler.go

```diff
@@ -169,18 +178,47 @@ func SignedTokenRedeemHandler(
 		} else {
 			logger.Trace().Msgf("request %s: validated", tokenRedeemRequestSet.Request_id)
 		}
-		if err := server.RedeemToken(verifiedIssuer, &tokenPreimage, request.Binding); err != nil {
-			logger.Error().Err(fmt.Errorf("request %s: token redemption failed: %w",
-				tokenRedeemRequestSet.Request_id, err)).
-				Msg("signed token redeem handler")
+		redemption, equivalence, err := server.CheckRedeemedTokenEquivalence(verifiedIssuer, &tokenPreimage, string(request.Binding))
+		if err != nil {
+			message := fmt.Sprintf("Request %s: Failed to check redemption equivalence", tokenRedeemRequestSet.Request_id)
+			return &utils.ProcessingError{
+				OriginalError:  err,
+				FailureMessage: message,
+				Temporary:      false,
+				KafkaMessage:   kafka.Message{},
+			}
+		}
+
+		// Continue if there is a duplicate
+		switch equivalence {
+		case cbpServer.IDEquivalence:
+			redeemedTokenResults = append(redeemedTokenResults, avroSchema.RedeemResult{
+				Issuer_name:     "",
+				Issuer_cohort:   0,
+				Status:          redeemDuplicateRedemptionID,
+				Associated_data: request.Associated_data,
+			})
+			continue
+		case cbpServer.IDAndAllValueEquivalence:
+			redeemedTokenResults = append(redeemedTokenResults, avroSchema.RedeemResult{
+				Issuer_name:     "",
+				Issuer_cohort:   0,
+				Status:          redeemDuplicateRedemptionAll,
```

Comment:
we discussed changing this to use `RedeemResultStatusIdempotent_redemption` in the generated avro go module rather than manually mirroring it in another enum